pilz_control/include/pilz_control/pilz_joint_trajectory_controller_impl.h

```diff
@@ -95,13 +95,23 @@ template <class SegmentImpl, class HardwareInterface>
 bool PilzJointTrajectoryController<SegmentImpl, HardwareInterface>::
 handleHoldRequest(std_srvs::TriggerRequest&, std_srvs::TriggerResponse& response)
 {
+  if(active_mode_ == Mode::HOLD)
+  {
+    response.message = "Already in hold mode";
+    response.success = true;
+    return true;
+  }
+
   std::lock_guard<std::mutex> lock(sync_mutex_);
```

Comment:
Why don't you lock_guard this shortcut? If a parallel service call to `/unhold` is pending but hasn't set the `active_mode_` yet, you might get a race condition.

Reply:
Correct. Will fix